pkg/gcp/client.go

```diff
@@ -250,3 +259,60 @@ func (c *gcpClient) UpdateWorkloadIdentityPoolOidcIdentityProvider(
 	}
 	return nil
 }
+
+// Uses HTTP requests directly to the Prometheus API endpoint
+// Makes GET request to: https://monitoring.googleapis.com/v1/projects/{project}/location/global/prometheus/api/v1/query
+func (c *gcpClient) QueryPrometheusTimeSeries(
+	ctx context.Context,
+	projectID string,
+	promqlQuery string,
+) (*PrometheusResponse, error) {
+	// Get default credentials
+	creds, err := google.FindDefaultCredentials(ctx, monitoringBaseURL)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get credentials")
+	}
+
+	// Create authenticated HTTP client
+	client := &http.Client{
+		Transport: &oauth2.Transport{
+			Source: creds.TokenSource,
+		},
+	}
```

Comment:
Having the highlighted code in this method means that a new client is made everytime this method is called. Instead, move this logic to the initializer of the gcpClient object. Call it monitoringClient, and then use this member of the object to perform the query.